ios/RNSScreenStack.mm

```diff
@@ -731,19 +735,43 @@ - (void)cancelTouchesInParent
   // item is close to an edge and we start pulling from edge we want the Touchable to be cancelled.
   // Without the below code the Touchable will remain active (highlighted) for the duration of back
   // gesture and onPress may fire when we release the finger.
+#ifdef RCT_NEW_ARCH_ENABLED
+  // On Fabric there is no view that exposes touchHandler above us in the view hierarchy, however it is still
+  // utilised. `RCTSurfaceView` should be present above us, which hosts `RCTFabricSurface` instance, which in turn
+  // hosts `RCTSurfaceTouchHandler` as a private field. When initialised, `RCTSurfaceTouchHandler` is attached to the
+  // surface view as a gestureRecognizer <- and this is where we can lay our hands on it.
   UIView *parent = _controller.view;
-  while (parent != nil && ![parent respondsToSelector:@selector(touchHandler)])
+  while (parent != nil && ![parent isKindOfClass:RCTSurfaceView.class]) {
     parent = parent.superview;
-  if (parent != nil) {
-#ifdef RCT_NEW_ARCH_ENABLED
-    RCTSurfaceTouchHandler *touchHandler = [parent performSelector:@selector(touchHandler)];
+  }
+
+  // This could be possible in modal context
+  if (parent == nil) {
+    return;
+  }
+
+  RCTSurfaceTouchHandler *touchHandler = nil;
+  // Experimentation shows that RCTSurfaceTouchHandler is the only gestureRecogniser registered here,
+  // so we should not be afraid of any performance hit here.
+  for (UIGestureRecognizer *recognizer in parent.gestureRecognizers) {
+    if ([recognizer isKindOfClass:RCTSurfaceTouchHandler.class]) {
+      touchHandler = static_cast<RCTSurfaceTouchHandler *>(recognizer);
+    }
+  }
+
+  [touchHandler rns_cancelTouches];
```

Comment:
Is it the default way of naming the extension (or however it is called) ?

Reply:
Yes it is, see Obj-C docs: [link](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/CustomizingExistingClasses/CustomizingExistingClasses.html) (section "Avoid Category Method Name Clashes")

Reply:
I think I'd go with rnscreens_cancelTouches since many libs can have a prefix of `rns`. wdyt?

Reply:
Makes sense for me. https://github.com/software-mansion/react-native-screens/pull/2131/commits/30cfc12bdb45107a140453dc33b27ab848a8fa9f